code/modules/reagents/chemistry/reagents/drinks_reagents.dm

```diff
@@ -751,6 +757,14 @@
 /datum/reagent/consumable/drink/fyrsskar_tears/on_mob_delete(mob/living/M)
 	if(isskrell(M))
 		REMOVE_TRAIT(M, TRAIT_ALCOHOL_TOLERANCE, id)
+	UnregisterSignal(M, COMSIG_AFTER_SPECIES_CHANGE)
+	return ..()
+
+/datum/reagent/consumable/drink/fyrsskar_tears/proc/on_species_change(mob/living/M)
+	if(!isskrell(M))
```

Comment:
```suggestion
	SIGNAL_HANDLER // COMSIG_AFTER_SPECIES_CHANGE
	if(!isskrell(M))
```
this macro and comment should be at the top of every proc you plan to register to that signal. this prevents certain code from getting into the path of signals where it might cause problems, like waiting on user input (the comment is just for the reader)